src/main/java/marquez/service/models/Version.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package marquez.service.models;
+
+import lombok.NonNull;
+import lombok.Value;
+
+@Value(staticConstructor = "of")
+public class Version<T> {
```

Comment:
_Minor_: I think it's ok to use the class `Version` as a wrapper for a `UUID` and not introduce the generic type `T` as a version will only be a `UUID`. I had originally thought of introducing the `Version` class as a way to distinguish between:

1.  A row `UUID` (_primary key_)
2. A dataset version or job version

Mind updating the class to wrap a `UUID`? Otherwise, code looks great!

Reply:
@wslulciuc, done. thanks!

Reply:
codcov is failing, I would add few test cases for Version class sometime this week

Reply:
Thanks, @ravikamaraj!

Reply:
@wslulciuc 
There was test failure, and is most likely due to circleci environmental issue. The same test is passing in my local. As I don't have permission to rerun the workflow, can you help to rerun circleci workflow

thanks